src/main/resources/io/jenkins/plugins/designlibrary/sample.js

```diff
@@ -0,0 +1,54 @@
+document.addEventListener("DOMContentLoaded", () =>{
+
+  function extractLanguageFromClassList(element) {
+    return Array.from(element.classList)
+      .filter(clazz => clazz.startsWith('language-'))
+      .map(clazz => clazz.replace('language-', ''));
+  }
+  const url = document.querySelector('head').dataset.rooturl
+
+  document.querySelectorAll('.sample-remote')
+    .forEach(element => {
+      const fileName = element.dataset.sample;
+      const componentName = window.location.href.match(/.+design-library\/(.+)$/)[1]
+
+      const fullUrl = `${url}/plugin/design-library/${componentName}${fileName}`;
+      fetch(fullUrl)
+        .then(response => response.text())
+        .then(text => {
+          const language = extractLanguageFromClassList(element);
+
+          if (language.length > 0) {
+            element.innerHTML = Prism.highlight(text, Prism.languages[language], language)
+          } else {
+            element.innerHTML = text
+          }
+        })
+    })
+
+  document.querySelectorAll('.language-java,.language-xml,.language-html,.language-css')
+    .forEach(element => {
+      const language = extractLanguageFromClassList(element);
+
+      if (language.length > 0) {
+        element.innerHTML = Prism.highlight(element.innerHTML, Prism.languages[language], language)
+      }
+    });
+
+  if (!navigator.canShare) {
+    document.querySelector('#button-share').style.display = "none"
+  }
+
+  document.querySelector('#button-share').addEventListener('click', async () => {
```

Comment:
Looks like html unit breaks on this syntax, https://github.com/jenkinsci/ui-samples-plugin/pull/14/checks?check_run_id=5716218242

missing formal parameter

may be related to https://github.com/HtmlUnit/htmlunit/issues/232